### DISCUSSION TITLE
[BUG] Manual Actions runs with `workflow_dispatch` always checks out a tree one commit out of date

### DISCUSSION THREAD

**Discussion by mbestavros** [community/community#26071] [Actions]
<p>I’m using Github Actions with my project for CI and automation, and I was particularly excited to use the newly-announced <code>workflow-dispatch</code> event for manual actions. However, I’ve encountered a pretty major bug in my time using it: any runs triggered by it seem to check out a tree that’s one commit older than the branch’s latest commit, even though the run says it’s using the latest commit.</p>
<p>See <a href="https://github.com/enarx/bot/runs/925151281?check_suite_focus=true" rel="nofollow noopener">here</a> for an example of the problem.</p>
<p>In this case, I’d submitted <a href="https://github.com/enarx/bot/commit/c0ffa8360ca3b2b18f30ae12fd4009dd85542966" rel="nofollow noopener">a patch</a> to fix a bug we’d uncovered in an Actions run. However, after my colleague initiated a manual run with <code>workflow_dispatch</code>, we ran into the exact same error as before – even down to the line number: the Actions log says the Python script failed on line 59, but as is visible in the diff for the commit that the manual run should be running on (commit <code>c0ffa83</code>), line 59 was changed to something completely different, and line 60 is what should have prompted the error, if there was one.</p>
<p>What’s even stranger is that the exact same code and the exact same commit ran perfectly once it hit a scheduled re-run. The only conclusion I can draw is that Actions’ <code>workflow_dispatch</code> trigger is somehow bugged.</p>

**Comment by dogboydog** [comment]
<p>I’m probably missing something, but which part of this workflow checks out the code from the rest of your repository? I’m not really familiar with nginx, but I don’t think running a workflow_dispatch or other workflow automatically checks out your whole repository, does it?  I have <span class="mention">@actions</span>/checkout in all of my workflows, but maybe you’re updating the files in some other way? How do they get into <code>/usr/local/bin/</code>?</p>
<p>Edit: I see you are pulling a container that copies some stuff into <code>/usr/local/bin/</code>. Are you sure  the version you’re pulling is the one you want? Sorry if I’m totally off base</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/mbestavros">@mbestavros</a>,</p>
<p>To confirm the commit info with ‘workflow_dispatch’, i created a simple yaml as below:</p>
<pre><code class="lang-auto">name: manual check
on: [workflow_dispatch]
jobs:
  job1:
    runs-on: ubuntu-latest
    steps:
      - name: checkout 
        uses: actions/checkout@v2
      - name: check the update content
        run: |
          cat test.txt
</code></pre>
<p>Edit the test.txt(commit <a href="https://github.com/weide-zhou/ticket34/commit/ff3db5f06e396ea4d6b726fd359679a97baa7859" rel="nofollow noopener"><code>ff3db5f</code></a>), and trigger manual workflow, confirmed the latest commit is included.<br>
<img src="https://user-images.githubusercontent.com/52772761/181095072-030d0ac1-e54f-4e35-8708-d23c767e3763.png" alt="image" data-base62-sha1="bHi1lYP57qHkhrIWwdCGBOGMyf9" width="518" height="252"></p>
<p>Hence, it seems that in your yaml code, you didn’t pull the correct content.<br>
You can simply add <code>uses: actions/checkout@v2</code> step in the workflow yaml, cat the file content for confirmation.</p>
<p>Thanks.</p>

**Comment by mbestavros** [comment, marked as answer]
<p><a class="mention" href="/weide-zhou">@weide-zhou</a> Thanks for your help! It turned out it wasn’t a bug with Actions, but rather my setup:</p>
<p>The scripts that were running with a delay were included as part of a container image that was only rebuilt when I pushed to master. So when I reran it on a different branch, I wasn’t actually using updated code. Apologies for the confusion!</p>
